src/Symfony/Component/DependencyInjection/Compiler/AnalyzeServiceReferencesPass.php

```diff
@@ -62,16 +62,22 @@ public function process(ContainerBuilder $container)
         $this->graph = $container->getCompiler()->getServiceReferenceGraph();
         $this->graph->clear();
         $this->lazy = false;
+        $this->definitions = $container->getDefinitions();
+        $this->aliases = $container->getAliases();
 
-        foreach ($container->getAliases() as $id => $alias) {
+        foreach ($this->aliases as $id => $alias) {
             $targetId = $this->getDefinitionId((string) $alias);
-            $this->graph->connect($id, $alias, $targetId, $this->getDefinition($targetId), null);
+            $this->graph->connect($id, $alias, $targetId, null !== $targetId ? $this->container->getDefinition($targetId) : null, null);
         }
 
-        parent::process($container);
+        try {
+            parent::process($container);
+        } finally {
+            $this->aliases = $this->definitions = array();
+        }
     }
 
-    protected function processValue($value, $isRoot = false)
+    protected function processValue($value, $isRoot = false, $inExpression = false)
```

Comment:
no `bool` typehint, as this is targetting 4.2 ? Unless you don't want to add a typehint while the other args are not typehinted (which would hurt merges) ?

Reply:
typehint added